robot/Cumulus/tests/browser/zpayment_allocations/enable_pmt_alloc_and_test.robot

```diff
@@ -0,0 +1,113 @@
+*** Settings ***
+
+Resource        robot/Cumulus/resources/NPSP.robot
+Suite Setup     Run keywords
+...             Open Test Browser
+...             Enable Payment Allocations
+...             Setup Test Data
+Suite Teardown  Run keywords
+...             Disable Payment Allocations
+...             Delete Records and Close Browser
+
+*** Test Cases ***
+
+Create Payment Allocations and Verify Opportunity Allocations Sync
+    Go To Record Home  &{opportunity}[Id]
+    Select Tab    Related
+    Load Related List    GAU Allocations
+    Click Link    &{payment}[Name]    
+    Select Window
+    Select Tab    Related
+    Verify Allocations    Payment Allocations
+    ...    &{def_gau}[Name]=$100.00
+    Click Related List Button    Payment Allocations    New
+    Populate Lookup Field    General Accounting Unit    &{gau}[Name]
+    Populate Field    Amount    40
+    Click Modal Button    Save
+    Wait Until Modal Is Closed
+    Verify Allocations    Payment Allocations
+    ...    &{def_gau}[Name]=$60.00
+    ...    &{gau}[Name]=$40.00     
+    Select Tab    Details
+    Click Link    &{opportunity}[Name]
+    Select Tab    Related
+    Load Related List    GAU Allocations
+    Verify Gau Allocations    GAU Allocations
+    ...    &{def_gau}[Name]=$60.00
+    ...    &{gau}[Name]=$40.00
+
+Update GAU Allocations and Verify Payment Allocations Sync
+    Click Special Related List Button  GAU Allocations    Manage Allocations
+    Choose Frame    Manage Allocations
+    Add GAU Allocation    Percent 0    60
+    Click Button    Save
+    Wait Until Page Does Not Contain    Add Row
+    Select Window
+    Load Related List    GAU Allocations
+    Verify Gau Allocations    GAU Allocations
+    ...    &{def_gau}[Name]=$40.00
+    ...    &{gau}[Name]=$60.00
+    # As a workaround for lightning cache issue, going to the payment record directly as it would reload the record
+    Go To Record Home    &{payment}[Id]
+    Select Tab    Related
+    Verify Allocations    Payment Allocations
+    ...    &{def_gau}[Name]=$40.00
+    ...    &{gau}[Name]=$60.00
+    
+***Keywords***
+
+Enable Payment Allocations
+    &{def_gau} =  API Create GAU    Name=default gau
+    Set Global Variable     &{def_gau}    &{def_gau}
+    Go To Setup Page    CustomSettings
+    Select Frame And Click Element    Custom Settings    custom_settings.link    Allocations Settings    Manage
+    Unselect Frame
+    Wait Until Loading Is Complete
+    Choose Frame    Custom Setting Allocations Settings
+    Wait Until Element Is Visible  text:Default Organization Level Value
+    Checkbox Status    Payment Allocations Enabled    Not Checked
+    Checkbox Status    Default Allocations Enabled    Not Checked
```

Comment:
The only differences I see for the keywords 'Enable payment allocations' and Disable payment allocations' are
- Checkbox statuses
- Populating the Default__c field

Can we not create just one keyword for these both and pass the status and default__c value as the parameters ? And you can  move the creation of &{def_gau} to 'Setup test data' keyword itself.

Reply:
I have to create gau as part of enabling as I need to provide gau id while enabling. And my other test can only be created after the GAU is enabled. I am adding GAU as part of enabling apart from checkbox and removing GAU as part of disabling. So I cannot combine these keywords